contracts: remove op-nft

tynes opened the pull request:
**Description**

Remove the `op-nft` package from the contracts package. None of these
contracts are maintained and can be moved to another repo if necessary.
The purpose of this commit is to reduce compilation time. Less files
to compile means less compilation time.

<!--
Contributions welcome! See https://github.com/ethereum-optimism/.github/blob/master/CONTRIBUTING.md
-->

tynes committed:
contracts: remove op-nft

Remove the `op-nft` package from the contracts package. None of these
contracts are maintained and can be moved to another repo if necessary.
The purpose of this commit is to reduce compilation time. Less files
to compile means less compilation time.

blmalone committed:
fix: semver lock step - ran pre-pr.

blmalone reviewed (approved):
nice, lgtm!

tynes commented:
Hey @smartcontracts, looks like we hit an edge case in the bash script for the heavy fuzz CI job

```
#!/bin/bash -eo pipefail
TEST_FILES=$(git diff origin/develop...HEAD --name-only -- './test/**/*.t.sol' | sed 's|packages/contracts-bedrock/||')
TEST_FILES=$(echo "$TEST_FILES" | circleci tests split --split-by=timings)
TEST_FILES=$(echo "$TEST_FILES" | sed 's|^test/||')
MATCH_PATH="./test/{$(echo "$TEST_FILES" | paste -sd "," -)}"
forge test --match-path "$MATCH_PATH"

Error reading historical timing data: file does not exist
Requested weighting by historical based timing, but they are not present. Falling back to weighting by name.
No tests match the provided pattern:
        match-path: `./test/{periphery/op-nft/AttestationStation.t.sol,periphery/op-nft/Optimist.t.sol,periphery/op-nft/OptimistAllowlist.t.sol,periphery/op-nft/OptimistInviter.t.sol}`
Error: 
No tests to run

Exited with code exit status 1
```

When a file is deleted, we need to be able to handle that case. Perhaps filter out before passing to foundry?

smartcontracts commented:
> Hey @smartcontracts, looks like we hit an edge case in the bash script for the heavy fuzz CI job
> 
> ```
> #!/bin/bash -eo pipefail
> TEST_FILES=$(git diff origin/develop...HEAD --name-only -- './test/**/*.t.sol' | sed 's|packages/contracts-bedrock/||')
> TEST_FILES=$(echo "$TEST_FILES" | circleci tests split --split-by=timings)
> TEST_FILES=$(echo "$TEST_FILES" | sed 's|^test/||')
> MATCH_PATH="./test/{$(echo "$TEST_FILES" | paste -sd "," -)}"
> forge test --match-path "$MATCH_PATH"
> 
> Error reading historical timing data: file does not exist
> Requested weighting by historical based timing, but they are not present. Falling back to weighting by name.
> No tests match the provided pattern:
>         match-path: `./test/{periphery/op-nft/AttestationStation.t.sol,periphery/op-nft/Optimist.t.sol,periphery/op-nft/OptimistAllowlist.t.sol,periphery/op-nft/OptimistInviter.t.sol}`
> Error: 
> No tests to run
> 
> Exited with code exit status 1
> ```
> 
> When a file is deleted, we need to be able to handle that case. Perhaps filter out before passing to foundry?

Good catch, thanks, I can open a PR tomorrow to filter out removed files

smartcontracts committed:
fix: use added/modified filter for heavy fuzz

Updates the heavy fuzz filter for CI to only check for added or
modified contracts, now excluding deleted or moved contracts.
